### PR TITLE
[LLM Runtime] Convert huggingface gptq model to jblas

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

detail description 
JIRA ticket: [NLPTOOLKIU-1049](https://jira.devtools.intel.com/browse/NLPTOOLKIU-1049)

- [x] Support llama2 gptq
- [x] Support neural_chat_v3_1 gptq w/o act order
- [x] Support neural_chat_v3_1 gptq with act order
- [x] update readme

## Expected Behavior & Potential Risk

```
from transformers import AutoTokenizer, TextStreamer
from intel_extension_for_transformers.transformers import AutoModelForCausalLM, WeightOnlyQuantConfig

# Download Hugging Face GPTQ model to local path
model_name = "PATH_TO_MODEL"  # local path to model
woq_config = WeightOnlyQuantConfig(use_gptq=True)
prompt = "Once upon a time, a little girl"

tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
inputs = tokenizer(prompt, return_tensors="pt").input_ids
streamer = TextStreamer(tokenizer)
model = AutoModelForCausalLM.from_pretrained(model_name, quantization_config=woq_config, trust_remote_code=True)
outputs = model.generate(inputs, streamer=streamer, max_new_tokens=300)
```

neural-chat-v3.1 w/o act order
```
<s> Once upon a time, a little girl named Alice was born. She was a curious child who loved to explore and learn about the world around her. As she grew older, she became more interest
ed in the mysteries of the universe.

One day, Alice decided to embark on a journey to find the answers to the mysteries of the universe. She packed her bags and set out on a quest to discover the secrets of the cosmos.

Alice traveled across the world, visiting various countries and learning about their cultures and traditions. She met many people along the way, each with their own unique stories and 
experiences.

As she traveled, Alice learned about the history of the universe, the formation of galaxies, and the mysteries of the stars. She studied the laws
```

llama2 gptq w/o act order
```
<s> Once upon a time, a little girl named Lily lived in a small village nestled in the rolling hills of the countryside. Lily was a curious and adventurous child, always eager to explore the world around her. She spent her days playing in the fields and forests, discovering new sights and sounds at every turn.
One day, as she wandered through the forest, Lily stumbled upon a hidden glade. The glade was filled with a dazzling array of wildflowers, and a babbling brook ran through the center of it. Lily was enchanted by the beauty of the glade, and she sat down on a rock to take in the sights and sounds around her.
As she sat there, Lily noticed a small, shimmering light hovering in the air above the brook. The light grew brighter and brighter, until it became a brilliant, glowing orb. Lily was both fascinated and a little scared by the orb, and she watched it with wide eyes.
Suddenly, the orb spoke to her in a soft, gentle voice. "Lily, I am the guardian of this forest," it said. "I have been watching over this place for many years, and I have seen many children come and go. But you, Lily, are different. You have a special spark within you, a spark that will help you on your journey."
Lily was amazed and thrilled by the talking orb's words. She had never heard of anything like this before, and she felt a surge of excitement and wonder.
The orb continued to speak, "Lily, you have a special gift, a gift that will help you on your journey. You have the power to heal and protect the creatures of this forest. Use your gift wisely, and always remember to respect and honor the creatures of this forest."
Lily was overjoyed by the orb's words, and she felt a deep sense of purpose and responsibility. She knew that she had a special role to play in the forest, and she was eager to begin her journey.
And so, Lily set off on her journey, using her gift to heal and protect the creatures of the forest. She spent her days exploring the forest, discovering new sights and sounds, and using her gift to help those in need. She was a kind and gentle presence in the forest, and the creatures of the forest looked up to her with gratitude and respect.
As the years passed, Lily grew in wisdom and power, and she became a beloved and respected member of the forest community. She continued to use her gift to help and protect the creatures of the forest, and she was a shining example of kindness and compassion.
And so, Lily lived out her days in the forest, surrounded by the creatures she had grown to love and respect. She was a true friend to the forest, and she will always be remembered as a kind and gentle soul.</s>
```


neural-chat-v3.1 w/ act order
```
 ### User:\ntell me about Intel### Assistant:\nIntel, short for Integrated Electronics Corporation, is an American multinational corporation and a global leader i
n the design and manufacture of semiconductor chips and computer hardware. Founded in 1968 by Gordon Moore and Robert Noyce, it's headquartered in Santa Clara, Ca
lifornia.                                                                                                                                                         
                                                                                                                                                                  
Intel's most significant products include microprocessors (CPUs), chipsets, motherboards, and other essential computing components. They have played a crucial rol
e in driving technological advancements throughout the years, shaping modern computing systems and contributing to innovations such as cloud computing, artificial
 intelligence, and the Internet of Things.                                                                                                                        
                                                                                                                                                                  
Throughout its history, Intel has developed numerous notable products and innovations, including:                                                                 
                                                                                                                                                                  
1. The original 4004 microprocessor in 1971 (the first commercial microprocessor)
2. The 8086 CPU in 1978 (the predecessor to the x86 architecture that powers most personal computers today)
3. The Pentium line of processors, beginning with the Pentium Pro in 1995 and leading up to current desktop and mobile processors
4. Advancements in chipsets, such as the Intel Centrino platform for wireless connectivity in laptops
5. Key contributions to the development of cloud computing and big data technologies
6. Artificial intelligence (AI) acceleration with products like the Nervana Neural Network Processor (NNP) and Habana Gaudi AI Training Processor
7. 3D XPoint non-volatile memory technology, which provides faster access times and increased storage capacity compared to traditional SSDs or DRAM.

In addition to its technological advancements, Intel has faced several challenges in recent years, such as the rise of mobile devices running on rival processors 
like ARM from companies like Apple and Samsung. As a result, the company has pivoted its focus towards becoming more involved in areas like IoT, cloud computing, 
data centers, autonomous driving, and new types of memory systems. [end of text]
```

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 1]
```diff
@@ -24,28 +24,71 @@ LLM Runtime is designed to provide the efficient inference of large language mod
...
-|[BLOOM-7B](https://huggingface.co/bigscience/bloomz-7b1)| ✅ | ✅ |  Latest |
-|[OPT-125m](https://huggingface.co/facebook/opt-125m), [OPT-350m](https://huggingface.co/facebook/opt-350m), [OPT-1.3B](https://huggingface.co/facebook/opt-1.3b), [OPT-13B](https://huggingface.co/facebook/opt-13b)| ✅ | ✅ |  Latest |
-|[ChatGLM-6B](https://huggingface.co/THUDM/chatglm-6b), [ChatGLM2-6B](https://huggingface.co/THUDM/chatglm2-6b)| ✅ | ✅ |  4.33.1 |
-|[Baichuan-13B-Chat](https://huggingface.co/baichuan-inc/Baichuan-13B-Chat), [Baichuan2-13B-Chat](https://huggingface.co/baichuan-inc/Baichuan2-13B-Chat)| ✅ | ✅ |  4.33.1 |
-|[Mistral-7B](https://huggingface.co/mistralai/Mistral-7B-v0.1)| ✅ | ✅ | 4.34.0 or newer |
-|[Qwen-7B](https://huggingface.co/Qwen/Qwen-7B-Chat), [Qwen-14b](https://huggingface.co/Qwen/Qwen-14B-Chat)| ✅ | ✅ |  Latest |
-
+| Model Name | INT8 | INT4 | GTPQ | Transformer Version |
+|---|:---:|:---:|:---:|:---:|
+|| ✅ | ✅ | ✅ | Latest |
```
Comment (a32543254) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:29]: here have display issue

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/README.md]: updated

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/215/

**Comment by VincyZhang** [comment]
GCC 11 and running fp8/nf4 https://inteltf-jenk.sh.intel.com/job/ITREX-cpp-graph-extension/217/

**Comment by VincyZhang** [comment]
@airMeng  @luoyu-intel @a32543254 @zhenwei-intel please check the result.

File: intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                        f"layers.{i}.attention_norm.weight", list_vars, f)
+        convert_fp32_tensor(f"model.layers.{i}.post_attention_layernorm.weight",
+                        f"layers.{i}.ffn_norm.weight", list_vars, f)
+
+
+    f.close()
+    print(f"Success! saved as {out_path}")
+
+if __name__ == '__main__':
+    main()
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:137]: "no new line" tag

File: intel_extension_for_transformers/llm/runtime/graph/core/layers/inner_product.cpp [hunk 3]
```diff
@@ -48,17 +48,22 @@ void JblasGemmCompF32(const int M, const int N, const int K, const float* A, con
...
-    auto reduceA = kernel.mProA.createStorage(M, K, B->mBlockSize);
     utils::GemmProblem gp(1, M, N, K, B->mBlockSize);
+    auto reduceA = kernel.mProA.createReduceStorage(M, K, B->mBlockSize);
+    auto reordA = kernel.mProA.createReorderStorage(M, K, B->mBlockSize);
     if (B->IsAsym()) {
       reduceA.assign(WorkSpace);
+      WorkSpace += reduceA.mSize;
+    }
+    if (B->ShfIndice()) {
+      reordA.assign(WorkSpace);
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/core/layers/inner_product.cpp:59]: if no reduce and only shuffle, shall
```
WorkSpace += reordA.mSize;
```

Comment (luoyu-intel) [reply intel_extension_for_transformers/llm/runtime/graph/core/layers/inner_product.cpp:59]: There is no storage left, so not necessary to shift the workspace pointer.

File: intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py (hunk 2 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py]: fixed

**Review by airMeng — APPROVED** [review]

**Review by a32543254 — APPROVED** [review]
LGTM

**Pull request merged**